Document composing filter conditions

raddevon opened the pull request.

scotttrinh reviewed (approved):
Love it!

elprans reviewed (commented):
While you are at it, it might be worth talking about optionals and using `e.any` instead of a tree of `e.op(.. 'or' )`, because that question popped up recently.

raddevon commented:
@scotttrinh Could you make one more pass on my additions per @elprans' suggestion? Added two more code blocks below what I had already added and brief explanations of each.

scotttrinh reviewed (commented):
Might as well also mention `e.all` == `and` while we are at it.

raddevon commented:
Good call, @scotttrinh. Done!

scotttrinh reviewed (approved):
Fabulous, thanks!!

themajashurka commented on docs/select.rst:
```diff
@@ -274,6 +274,132 @@ params object. This should correspond to a boolean expression.
+    title: true,
+    filter: e.all(
+      e.set(
+        e.op(movie.title, "ilike", "captain%")
```
Don't these lines need a comma at the end?

raddevon replied:
Yes, I believe they do. Thank you for catching it and for letting us know!

raddevon replied:
https://github.com/edgedb/edgedb-js/pull/744
